pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -753,137 +724,167 @@ func (r *ReconcileChaosEngine) removeChaosRunner(engine *chaosTypes.EngineInfo,
+func (r *ReconcileChaosEngine) reconcileForRestart(request reconcile.Request) (reconcile.Result, error) {
+	err := r.forceRemoveChaosResources(engine, request)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
-	if err := r.initEngineStatus(engine); err != nil {
-		return err
+	if err = r.updateEngineForRestart(engine); err != nil {
+		return reconcile.Result{}, nil
+	}
+	return reconcile.Result{}, nil
+
+}
+
+// initEngine initilizes Chaos Engine, and add a finalizer to it.
+func (r *ReconcileChaosEngine) initEngine(engine *chaosTypes.EngineInfo) error {
+	if engine.Instance.Spec.EngineState == "" {
+		engine.Instance.Spec.EngineState = litmuschaosv1alpha1.EngineStateActive
+	}
+	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateActive && engine.Instance.Status.EngineStatus == "" {
+		engine.Instance.Status.EngineStatus = litmuschaosv1alpha1.EngineStatusInitialized
+	}
+	if engine.Instance.Status.EngineStatus == litmuschaosv1alpha1.EngineStatusInitialized {
+		if engine.Instance.ObjectMeta.Finalizers == nil {
+			engine.Instance.ObjectMeta.Finalizers = append(engine.Instance.ObjectMeta.Finalizers, finalizer)
+			r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "ChaosEngineInitialized", "%s created successfully", engine.Instance.Name+"-runner")
+			if err := r.client.Update(context.TODO(), engine.Instance, &client.UpdateOptions{}); err != nil {
+				return err
+			}
+		}
 	}
 	return nil
 }
 
-func (r *ReconcileChaosEngine) initEngineStatus(engine *chaosTypes.EngineInfo) error {
-	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateActive && engine.Instance.Status.EngineStatus != litmuschaosv1alpha1.EngineStatusCompleted {
+// reconcileForCreationAndRunning reconciles for Chaos execution of Chaos Engine
+func (r *ReconcileChaosEngine) reconcileForCreationAndRunning(engine *chaosTypes.EngineInfo, reqLogger logr.Logger) (reconcile.Result, error) {
 
-		err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusInitialized)
-		if err != nil {
-			return err
-		}
+	err := r.validateAnnontatedApplication(engine)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	//Check if the engineRunner pod already exists, else create
+	err = r.checkEngineRunnerPod(engine, reqLogger)
+	if err != nil {
+		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to get chaos resources")
+		return reconcile.Result{}, err
+	}
+	// If monitoring is set to true,
+	// Define an engineMonitor pod which is secondary-resource #2 and
+	// Define an engineMonitor service which is secondary-resource #3
+	// in the same namespace as CR
+	err = r.checkMonitoringPodAndSvc(reqLogger)
+	if err != nil {
+		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to get chaos resources")
+		return reconcile.Result{}, err
+	}
 
-		err = r.addFinalizerToEngine(engine, finalizer)
+	isCompleted := r.checkRunnerPodCompletedStatus(engine)
```

Comment:
So what used to be a separate sub-reconcile func is now added as a step in the reconcileForCreationAndRunning. Which is good.

Reply:
Yes, @ksatchit